### PR TITLE
Add router into vue app generated

### LINKED ISSUE #1018: Frontend: Add router into Vue app generated
As already implemented into the webapp generator, the point is to generate by default the router into a vue app

### PR DESCRIPTION
**Pull request by seraphinandrieux**
- Fix #1018

### CODE REVIEW

File: src/main/resources/generator/client/vue/webapp/app/main.ts.mustache [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { createApp } from 'vue';
 import App from './common/primary/app/App.vue';
+// jhipster-needle-import
```
Comment (pascalgrimaud) [inline src/main/resources/generator/client/vue/webapp/app/main.ts.mustache:3]: could you rename to something like: `// jhipster-needle-main-ts-import`

File: src/main/resources/generator/client/vue/webapp/app/main.ts.mustache [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
 
-createApp(App).mount('#app');
+const app = createApp(App);
+// jhipster-needle-provider
```
Comment (pascalgrimaud) [inline src/main/resources/generator/client/vue/webapp/app/main.ts.mustache:6]: could you rename to something like: `// jhipster-needle-main-ts-provider`

File: src/test/java/tech/jhipster/lite/generator/client/vue/core/application/VueAssert.java [hunk 3]
```diff
@@ -39,6 +40,11 @@ public static void assertRootFiles(Project project) {
     assertFileExist(project, "src/main/webapp/app/main.ts");
   }
 
+  public static void assertRouterFiles(Project project) {
+    assertFileExist(project, "src/main/webapp/app/router/router.ts");
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/client/vue/core/application/VueAssert.java:44]: can use constant MAIN_WEBAPP with getPath

File: src/test/java/tech/jhipster/lite/generator/client/vue/core/application/VueAssert.java [hunk 4]
```diff
@@ -39,6 +40,11 @@ public static void assertRootFiles(Project project) {
...
+    assertFileExist(project, "src/test/javascript/spec/router/Router.spec.ts");
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/client/vue/core/application/VueAssert.java:45]: maybe we could add a constant here for src/test/javascript/spec

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by seraphinandrieux** [comment]
bounty claimed : https://opencollective.com/generator-jhipster/expenses/68143

**Comment by pascalgrimaud** [comment]
@seraphinandrieux : approved, thanks a lot for your work here

**Pull request merged**